File: .archive/frontend_v1.1_backup_20251121_134032/src/types/wiki.ts
/**
 * Wiki API Type Definitions
 *
 * Matches the Pydantic schemas from src/finagent/api/schemas/wiki.py
 */

// ============================================================================
// Category Types
// ============================================================================

export interface CategorySummary {
  id: number
  name: string
  type: 'authority' | 'institution' | 'violation' | 'doc_type'
  document_count: number
  description?: string
  keywords: string[]
}

export interface CategoryDetail extends CategorySummary {
  metadata: Record<string, any>
  created_at: string
  updated_at: string
}

export interface CategoryTree {
  type: 'authority' | 'institution' | 'violation' | 'doc_type'
  total_count: number
  categories: CategorySummary[]
}

// ============================================================================
// Document Types
// ============================================================================

export interface DocumentSummary {
  doc_id: string
  filename: string
  document_type?: string
  issuing_authority?: string
  date?: string
  related_institutions: string[]
  violation_types: string[]
  extraction_confidence?: number
}

export interface RelatedDocument {
  doc_id: string
  filename: string
  relationship_type: 'related' | 'temporal' | 'citation'
  strength: number // 0.0-1.0
  reason?: string
}

export interface DocumentDetail {
  doc_id: string
  filename: string
  file_path: string
  document_type?: string
  issuing_authority?: string
  date?: string
  case_number?: string
  related_institutions: string[]
  violation_types: string[]
  penalty_amount?: string
  keywords: string[]
  extraction_confidence?: number
  full_content?: string
  chunk_count: number
  indexed: boolean
  file_size?: number
  created_at: string
  updated_at: string
  related_documents: RelatedDocument[]
}

export interface DocumentList {
  total: number
  offset: number
  limit: number
  documents: DocumentSummary[]
}

// ============================================================================
// Statistics Types
// ============================================================================

export interface DocumentStats {
  total_documents: number
  with_metadata: number
  without_metadata: number
  by_type: Record<string, number>
  by_authority: Record<string, number>
  avg_confidence?: number
}

export interface TimelineDataPoint {
  period: string // e.g., "2020", "2020-01", "2020-01-15"
  count: number
  label?: string
}

export interface TimelineStats {
  granularity: 'year' | 'month' | 'day'
  data: TimelineDataPoint[]
  total_count: number
  date_range: {
    earliest?: string
    latest?: string
  }
}

export interface TopEntity {
  name: string
  count: number
  percentage: number
}

export interface TopEntitiesStats {
  institutions: TopEntity[]
  violations: TopEntity[]
  authorities: TopEntity[]
}

// ============================================================================
// Wiki Overview
// ============================================================================

export interface WikiOverview {
  total_documents: number
  total_categories: number
  categories_by_type: Record<string, number>
  document_stats: DocumentStats
  recent_documents: DocumentSummary[]
  top_entities: TopEntitiesStats
  generated_at?: string
}

// ============================================================================
// Search Types
// ============================================================================

export interface SearchFilters {
  document_type?: string
  authority?: string
  institution?: string
  violation?: string
  date_from?: string
  date_to?: string
  min_confidence?: number
}

export interface SearchResult {
  doc_id: string
  filename: string
  document_type?: string
  relevance_score: number
  snippet?: string
  highlights: string[]
}

export interface SearchResponse {
  query: string
  total_results: number
  offset: number
  limit: number
  results: SearchResult[]
  filters_applied?: SearchFilters
}

// ============================================================================
// Operation Response Types
// ============================================================================

export interface WikiRebuildResponse {
  success: boolean
  rebuild_time_ms: number
  categories_created: number
  documents_categorized: number
  relationships_detected: number
  statistics_updated: boolean
  message: string
}

export interface StatsRefreshResponse {
  success: boolean
  statistics_updated: boolean
  cached_at: string
  message: string
}

export interface ErrorResponse {
  detail: string
  error_type?: string
  error_code?: string
}

File: src/finagent/api/schemas/wiki.py
"""
Wiki API Response Schemas

Pydantic models for wiki API endpoints.
"""

from typing import Any

from pydantic import BaseModel, Field

# ============================================================================
# Category Schemas
# ============================================================================


class CategorySummary(BaseModel):
    """Summary of a single category."""

    id: int
    name: str
    type: str  # authority, institution, violation, doc_type
    document_count: int
    description: str | None = None
    keywords: list[str] = Field(default_factory=list)


class CategoryDetail(BaseModel):
    """Detailed category information with metadata."""

    id: int
    name: str
    type: str
    document_count: int
    description: str | None = None
    keywords: list[str] = Field(default_factory=list)
    metadata: dict[str, Any] = Field(default_factory=dict)
    created_at: str | None = None
    updated_at: str | None = None


class CategoryTree(BaseModel):
    """Hierarchical category tree."""

    type: str  # authority, institution, violation, doc_type
    total_count: int
    categories: list[CategorySummary]


# ============================================================================
# Document Schemas
# ============================================================================


class DocumentSummary(BaseModel):
    """Brief document information for lists."""

    doc_id: str
    filename: str
    document_type: str | None = None
    issuing_authority: str | None = None
    date: str | None = None
    related_institutions: list[str] = Field(default_factory=list)
    violation_types: list[str] = Field(default_factory=list)
    extraction_confidence: float | None = None


class DocumentDetail(BaseModel):
    """Full document metadata."""

    doc_id: str
    filename: str
    file_path: str
    document_type: str | None = None
    issuing_authority: str | None = None
    date: str | None = None
    case_number: str | None = None
    related_institutions: list[str] = Field(default_factory=list)
    violation_types: list[str] = Field(default_factory=list)
    penalty_amount: str | None = None
    keywords: list[str] = Field(default_factory=list)
    extraction_confidence: float | None = None
    full_content: str | None = None
    chunk_count: int = 0
    indexed: bool = False
    file_size: int | None = None
    created_at: str | None = None
    updated_at: str | None = None
    related_documents: list["RelatedDocument"] = Field(default_factory=list)


class RelatedDocument(BaseModel):
    """Related document with relationship strength."""

    doc_id: str
    filename: str
    relationship_type: str  # related, temporal, citation
    strength: float  # 0.0-1.0
    reason: str | None = None


class DocumentList(BaseModel):
    """Paginated document list."""

    total: int
    offset: int
    limit: int
    documents: list[DocumentSummary]


# ============================================================================
# Statistics Schemas
# ============================================================================


class DocumentStats(BaseModel):
    """Overall document statistics."""

    total_documents: int
    with_metadata: int
    without_metadata: int
    by_type: dict[str, int] = Field(default_factory=dict)
    by_authority: dict[str, int] = Field(default_factory=dict)
    avg_confidence: float | None = None


class TimelineDataPoint(BaseModel):
    """Single data point for timeline."""

    period: str  # e.g., "2020", "2020-01", "2020-01-15"
    count: int
    label: str | None = None


class TimelineStats(BaseModel):
    """Timeline statistics."""

    granularity: str  # year, month, day
    data: list[TimelineDataPoint]
    total_count: int
    date_range: dict[str, str | None] = Field(
        default_factory=lambda: {"earliest": None, "latest": None}
    )


class TopEntity(BaseModel):
    """Top entity (institution or violation) with count."""

    name: str
    count: int
    percentage: float


class TopEntitiesStats(BaseModel):
    """Top entities statistics."""

    institutions: list[TopEntity]
    violations: list[TopEntity]
    authorities: list[TopEntity]


# ============================================================================
# Wiki Overview Schemas
# ============================================================================


class WikiOverview(BaseModel):
    """Complete wiki overview."""

    total_documents: int
    total_categories: int
    categories_by_type: dict[str, int] = Field(default_factory=dict)
    document_stats: DocumentStats
    recent_documents: list[DocumentSummary] = Field(default_factory=list)
    top_entities: TopEntitiesStats
    generated_at: str | None = None


# ============================================================================
# Search Schemas
# ============================================================================


class SearchFilters(BaseModel):
    """Search filter parameters."""

    document_type: str | None = None
    authority: str | None = None
    institution: str | None = None
    violation: str | None = None
    date_from: str | None = None
    date_to: str | None = None
    min_confidence: float | None = None


class SearchResult(BaseModel):
    """Single search result."""

    doc_id: str
    filename: str
    document_type: str | None = None
    relevance_score: float
    snippet: str | None = None
    highlights: list[str] = Field(default_factory=list)


class SearchResponse(BaseModel):
    """Search results response."""

    query: str
    total_results: int
    offset: int
    limit: int
    results: list[SearchResult]
    filters_applied: SearchFilters | None = None


# ============================================================================
# Operation Response Schemas
# ============================================================================


class WikiRebuildResponse(BaseModel):
    """Wiki rebuild operation response."""

    success: bool
    rebuild_time_ms: int
    categories_created: int
    documents_categorized: int
    relationships_detected: int
    statistics_updated: bool
    message: str


class StatsRefreshResponse(BaseModel):
    """Statistics refresh response."""

    success: bool
    statistics_updated: bool
    cached_at: str
    message: str


class ErrorResponse(BaseModel):
    """Standard error response."""

    detail: str
    error_type: str | None = None
    error_code: str | None = None


# Update forward references
DocumentDetail.model_rebuild()
